### PR TITLE
refactored tests.py

### PR DESCRIPTION
**Pull request by danielfmiranda**
# Description

Related PRs/issues: #9791

As mentioned in the related ticket above, the `tests.py` file included various unrelated tests.

I have went through them, and for the following reasons refactored them like so:

**Testing for missing migrations**
Since this has to do with the app as a whole, these tests have been moved to `tests/test_migrations.py`

**Testing middleware**
These tests have been moved to `utility/tests/test_middleware`

**Testing redirect decorator**
These tests have been moved to `utility/tests/test_redirects`

**Testing users and groups**
These tests have been moved to `management/tests/test_delete_non_staff.py`

**Testing PNI airtable connection**
Since this has to do with PNI, these tests have been moved to `networkapi/wagtailpages/tests/buyersuide/test_airtable.py`

**Testing utilities**
These utilities were all defined within the `wagtailpages` directory, so the tests for them have been moved to `wagtailpages/tests/test_utils.py`

### CODE REVIEW

File: network-api/networkapi/tests/test_middleware.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+from unittest.mock import MagicMock
+
+from django.test import TestCase
+
+from networkapi.utility.middleware import ReferrerMiddleware, XRobotsTagMiddleware
```
Comment (tbrlpld) [inline network-api/networkapi/tests/test_middleware.py:5]: Considering that the classes under test are located in the `utility` app, have you considered moving the tests into that app as well (i.e. `network-api/networkapi/utility/tests/test_middleware`)?

File: network-api/networkapi/tests/test_redirects.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
+from unittest import skip
+
+from django.test import RequestFactory, TestCase
+from wagtail.core.models import Site
+from wagtail_factories import SiteFactory
+
+from networkapi.utility.redirects import redirect_to_default_cms_site
```
Comment (tbrlpld) [inline network-api/networkapi/tests/test_redirects.py:7]: Considering that the function under test is located in the utility app, have you considered moving the tests into that app as well (i.e. `network-api/networkapi/utility/tests/test_redirects`)?

File: network-api/networkapi/tests/test_users.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
+from django.contrib.auth.models import Group, User
+from django.core.management import call_command
+from django.test import TestCase
+
+
```
Comment (tbrlpld) [inline network-api/networkapi/tests/test_users.py:5]: Considering that these tests are for a management command that is defined in the `network-api/networkapi/management` app, have you considered adding moving this test module into that app as well (e.g. `network-api/networkapi/management/tests/test_delete_non_staff.py`)?

File: network-api/networkapi/wagtailpages/tests/test_utils.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+    def test_parse_accept_lang_header_returns_iso_3166_language(self):
+        self.assertEqual(
+            parse_accept_lang_header("en-GB,en;q=0.5"),
+            (("en-GB", 1.0), ("en", 0.5)),
+        )
+
+
+class TestDjangoTranslationUtilityOverrides(TestCase):
+
```
Comment (tbrlpld) [inline network-api/networkapi/wagtailpages/tests/test_utils.py:81]:
```suggestion
```

Comment (tbrlpld) [reply network-api/networkapi/wagtailpages/tests/test_utils.py:81]: Not needed empty line

File: network-api/networkapi/wagtailpages/tests/test_utils.py [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+        new_image = create_wagtail_image(
+            self.image_path,
+            image_name="fake teddy.jpg",
+            collection_name=new_collection_name,
+        )
+        self.assertEqual(new_image.collection.name, new_collection_name)
+
+
+class TestLanguageUtilities(TestCase):
```
Comment (tbrlpld) [inline network-api/networkapi/wagtailpages/tests/test_utils.py:64]: **Kudos** Thanks for fixing these class names 👍

**Review by tbrlpld — COMMENTED** [review]
Thanks for this @danielfmiranda This is coming a long nicely. Just left some small questions regarding the location of some of the test modules.

File: network-api/networkapi/tests/test_middleware.py (hunk 1 above)
Comment (danielfmiranda) [reply network-api/networkapi/tests/test_middleware.py:5]: Updated!

File: network-api/networkapi/tests/test_redirects.py (hunk 2 above)
Comment (danielfmiranda) [reply network-api/networkapi/tests/test_redirects.py:7]: Updated!

File: network-api/networkapi/tests/test_users.py (hunk 3 above)
Comment (danielfmiranda) [reply network-api/networkapi/tests/test_users.py:5]: Hi @tbrlpld this move makes sense to me also **kudos** I think that the rename is a great idea considering that all of these tests are testing the `delete_non_staff` command!

File: network-api/networkapi/wagtailpages/tests/test_utils.py (hunk 4 above)
Comment (danielfmiranda) [reply network-api/networkapi/wagtailpages/tests/test_utils.py:81]: Updated!

**Review by tbrlpld — APPROVED** [review]

**Pull request merged**